src/Components/CardboardListCallout/CardboardListCallout.tsx

```diff
@@ -124,12 +128,14 @@ const CardboardListCallout = <T extends unknown>({
 
                 {primaryActionProps && (
                     <PrimaryButton
-                        data-testid={dataButtonTestId}
+                        {...(dataButtonTestId && {
+                            'data-testid': dataButtonTestId
+                        })}
                         styles={cardboardListCalloutPrimaryButtonStyles}
                         onClick={() =>
                             primaryActionProps.onPrimaryActionClick(searchText)
                         }
-                        disabled={primaryActionProps.disabled}
+                        disabled={primaryActionProps.disabled ? true : false}
```

Comment:
nit: you can also just pass `!!primaryActionProps.disabled` if you want to convert it to a boolean.